### PR TITLE
stm32cube: Update STM32Cube packages

### PR DESCRIPTION
**Pull request by ajarmouni-st**
New batch of stm32cube package updates:
  - update stm32f1 to cube version V1.8.5
  - update stm32f7 to cube version V1.17.1
  - update stm32h7 to cube version V1.11.1
  - update stm32l4 to cube version  V1.18.0
  - update stm32u5 to cube version  V1.4.0
  - update stm32wba to cube version  V1.2.0
  - update stm32wb to cube version V1.18.0

An update of lib/stm32/stm32wb package to version V1.18.0

### CODE REVIEW

File: stm32cube/stm32h7xx/README [hunk 1]
```diff
@@ -69,8 +65,8 @@ Patch List:
      ST Internal Reference: 136825
 
    *Fix to use transmit timestamp when it is valid
-    Read timestamp only when status flag is set in Tx descriptor.
-    Fire transmit timestamp callbacks only when there is a valid
+Status:
+   version v1.11.1
```
Comment (erwango) [inline stm32cube/stm32h7xx/README:69]: So, was it fixed in the end ?

File: stm32cube/stm32wbaxx/CMakeLists.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2023 STMicroelectronics
+# Copyright (c) 2020 STMicroelectronics
```
Comment (erwango) [inline stm32cube/stm32wbaxx/CMakeLists.txt:1]: 2023 should be kept

**Review by erwango — CHANGES_REQUESTED** [review]

File: stm32cube/stm32h7xx/README [hunk 3]
```diff
@@ -69,8 +65,8 @@ Patch List:
      ST Internal Reference: 136825
 
    *Fix to use transmit timestamp when it is valid
-    Read timestamp only when status flag is set in Tx descriptor.
```
Comment (FRASTM) [inline stm32cube/stm32h7xx/README:72]: if related to ST  Internal Reference: 157129 it is not yet fixed (see ST Internal Reference: 157128)

**Review by FRASTM — COMMENTED** [review]
else LGTM

**Review by erwango — APPROVED** [review]

**Review by FRASTM — APPROVED** [review]

**Pull request merged**